app/src/ui/repository-settings/repository-settings.tsx

```diff
@@ -18,17 +18,32 @@ import { OkCancelButtonGroup } from '../dialog/ok-cancel-button-group'
 import { ForkSettings } from './fork-settings'
 import { ForkContributionTarget } from '../../models/workflow-preferences'
 import { enableForkSettings } from '../../lib/feature-flag'
+import { GitConfigLocation, GitConfig } from './git-config'
+import {
+  getConfigValue,
+  getGlobalConfigValue,
+  removeConfigValue,
+  setConfigValue,
+} from '../../lib/git/config'
+import {
+  gitAuthorNameIsValid,
+  InvalidGitAuthorNameMessage,
+} from '../lib/identifier-rules'
+import { Account } from '../../models/account'
 
 interface IRepositorySettingsProps {
+  readonly initialSelectedTab?: RepositorySettingsTab
   readonly dispatcher: Dispatcher
   readonly remote: IRemote | null
   readonly repository: Repository
+  readonly repositoryAccount: Account | null
   readonly onDismissed: () => void
 }
 
-enum RepositorySettingsTab {
+export enum RepositorySettingsTab {
   Remote = 0,
   IgnoredFiles,
+  GitConfig,
```

Comment:
I added this tab before `ForkSettings` because `ForkSettings` is optional (only shows up for forks). Adding a new non-optional tab after an optional tabs would require us some special treatment of tab indices.